### PR TITLE
[TASK] Add Extbase DebuggerUtility and Fluid Debug ViewHelper. Changed Order of contents

### PR DESCRIPTION
**Pull request by Marlineck**

### CODE REVIEW

**Comment by garvinhicking** [comment]
Thanks for contributing this!
I cannot currently review this, since I am mainly on mobile this week. It is sadly hard to review changes when sections are moved AND also changes are made, because then I cannot see what may have changed in a moved section.
For that I would need multiple windows/screens.

Tl;dr: can check this next week :)

**Comment by linawolf** [comment]
Thanks for your suggestions. Whenever you introduce or change a headline that didnt have one yet, please introduce an anchor: https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/Reference/ReStructuredText/Links/Anchors.html

Do you need support in doing so?

I will do some more reviews now and might have additional suggestions

File: Documentation/ApiOverview/Debugging/Index.rst [hunk 1]
```diff
@@ -10,49 +10,36 @@ Debugging
...
-First of all: best practice to debug PHP code is using an IDE (like `PhpStorm`_
-or `Visual Studio Code`_) with `Xdebug`_. Advantages are:
-
-*   You can investigate the values of variables and class properties when
-    setting a breakpoint.
-*   You can move forward line by line and see how the variables and properties
-    change.
-*   You can also step into calling methods and functions.
-*   You see the calling code as a stack trace.
+TYPO3 backend debug mode
```
Comment (linawolf) [inline Documentation/ApiOverview/Debugging/Index.rst:20]:
```suggestion
.. _examples-debug-backend:

TYPO3 backend debug mode
```

File: Documentation/ApiOverview/Debugging/Index.rst [hunk 2]
```diff
@@ -10,49 +10,36 @@ Debugging
...
+JavaScript or CSS resources.
 
-..  index::
-    GlobalDebugFunctions
-    DebugUtility
-    TYPO3_CONF_VARS; SYS devIPmask
 
-`DebugUtility::debug()`
------------------------
+DebugUtility::debug()
```
Comment (linawolf) [inline Documentation/ApiOverview/Debugging/Index.rst:41]:
```suggestion
.. _examples-debug-utility:

DebugUtility::debug()
```

File: Documentation/ApiOverview/Debugging/Index.rst [hunk 3]
```diff
@@ -75,22 +62,68 @@ debugging tools.
...
     TYPO3_CONF_VARS; BE debug
 
-TYPO3 backend debug mode
-========================
 
-To display additional debug information in the backend, set
-:ref:`$GLOBALS['TYPO3_CONF_VARS']['BE']['debug'] <typo3ConfVars_be_debug>`
-in :file:`config/system/settings.php` and log in with an administrator
-account.
+Extbase DebuggerUtility
```
Comment (linawolf) [inline Documentation/ApiOverview/Debugging/Index.rst:66]:
```suggestion
.. _examples-debug-extbase-utility:

Extbase DebuggerUtility
```

File: Documentation/ApiOverview/Debugging/Index.rst [hunk 4]
```diff
@@ -75,22 +62,68 @@ debugging tools.
...
+**Example:**
+
+:php:`\TYPO3\CMS\Extbase\Utility\DebuggerUtility::var_dump($myVariable);`
+
+You can also use the Extbase DebuggerUtility to debug SQL Querys for example. To do so, put the following code snippet before the execute function of your SQL query:
+
+:php:`\TYPO3\CMS\Extbase\Utility\DebuggerUtility::var_dump($queryBuilder->getSQL());`
+
+
+Fluid Debug ViewHelper
```
Comment (linawolf) [inline Documentation/ApiOverview/Debugging/Index.rst:80]:
```suggestion
.. _examples-debug-fluid:

Fluid Debug ViewHelper
```

File: Documentation/ApiOverview/Debugging/Index.rst [hunk 5]
```diff
@@ -75,22 +62,68 @@ debugging tools.
...
+To display all available variables in your Fluid template, you can use the _all placeholder:
+
+:html:`<f:debug>{_all}</f:debug>`
+
+..  note::
+    If you are debugging in a Fluid partial or a Fluid section, make sure that all variables you want to analyse are passed (defined in the arguments attribute of the render tag).
+
+Get more information in the Fluid ViewHelper Reference :ref:`t3viewhelper:typo3-fluid-debug`
+
+
```
Comment (linawolf) [inline Documentation/ApiOverview/Debugging/Index.rst:99]:
```suggestion

.. _examples-debug-xdebug:
```

**Review by linawolf — DISMISSED** [review]
Thanks for these valuable additions

**Review by linawolf — APPROVED** [review]
Thanks for your changes, sorry it took a while

**Pull request merged**